QualityControl/docker-compose.dev.yml

```diff
@@ -0,0 +1,17 @@
+services:
+  application:
+    build:
+      context: .
+      target: development
+    environment:
+      NODE_ENV: development
+    ports:
+      - "8080:8080"
+    container_name: qcg-application-dev
+    volumes:
+      # Mount the project into the container
+      - .:/usr/src/app:cached
+      # Ignore local node_modules
+      - /usr/src/app/node_modules
```

Comment:
Could you please explain that? Especially:
- Why `:cached` ?
- Why ignore local node_modules?